model/metric.go

```diff
@@ -79,3 +83,19 @@ func (m Metric) Fingerprint() Fingerprint {
 func (m Metric) FastFingerprint() Fingerprint {
 	return LabelSet(m).FastFingerprint()
 }
+
+// IsValidMetricName returns true iff name matches the pattern of MetricNameRE.
+func IsValidMetricName(n LabelValue) bool {
+	if len(n) == 0 {
+		return false
+	}
+	if !((n[0] >= 'a' && n[0] <= 'z') || (n[0] >= 'A' && n[0] <= 'Z') || n[0] == '_' || n[0] == ':') {
+		return false
+	}
+	for _, b := range n[1:] {
+		if !((b >= 'a' && b <= 'z') || (b >= 'A' && b <= 'Z') || b == '_' || b == ':' || (b >= '0' && b <= '9')) {
```

Comment:
The "more checks" only kick in for numbers, so pretty rarely. With the amount of replicated code (and potential of unintentional deviations between L92 and L96), I'd vote pretty clearly for the one line version.